GTC-2874 Step function and placeholder lambdas for OTF list analysis

danscales opened the pull request:
GTC-2874 Step function and placeholder lambdas for OTF list analysis

Used distributed map step, modified analysis, and various permissions from Justin's sample PR for using distributed map.

Still to be tested and have the preprocessing/aggregation lambdas filled in.

danscales committed:
GTC-2874 Step function and placeholder lambdas for OTF list analysis

Used distributed map step, modified analysis, and various permissions
from Justin's sample PR for using distributed map.

Still to be tested and have the preprocessing/aggregation lambdas filled
in.

danscales committed:
Have preprocessing lambda return some testing paths

And use a ResultSelector, not OutputPath for the Preprocessing output.

danscales committed:
Fix bucket/key values to be top-level bucket and path

danscales committed:
Change OutputPath to ResultPath for distribute map operation.

danscales committed:
Access data_environment, not environment in analysis lambda

jterry64 commented on lambdas/list_tiled_analysis/src/lambda_function.py:
```diff
@@ -0,0 +1,49 @@
+
+
+@xray_recorder.capture("Tiled Analysis")
+def handler(event, context):
```
This is because it's a different input, right? Would it be possible to pull out the common code into a function and have the handlers just pull out the parameters from the input?

danscales replied:
I think you're talking about the fact that we that the list_tiled_analysis lambda and the tiled_analysis lambda and they are fairly similar?  As you say, it is because the inputs are slightly different (we need to decode the WKB, we need to deal with the fid, no format arg, etc.)  Most of the common code is already shared in the calls to AnalysisTiler() and tiler.execute().  I'll take a look, but I'm not sure if it is really worth sharing any more code, given the remaining code is mostly arg and results handling.

jterry64 commented on terraform/lambdas.tf:
```diff
@@ -94,4 +94,101 @@ resource "aws_lambda_function" "raster_analysis_fanout" {
   }
-}
\ No newline at end of file
+}
```
thanks for adding these!

solomon-negusse committed:
WIP: aggregating outputs

solomon-negusse committed:
fix s3 client invocation

solomon-negusse committed:
debug

solomon-negusse committed:
debug

danscales committed:
Change from result_as_csv() to result_as_dict() in list_tiled_analysis

solomon-negusse committed:
return results as json

solomon-negusse committed:
write out aggregated result

solomon-negusse committed:
decode nested json

solomon-negusse committed:
fix combined variable

solomon-negusse committed:
result as list of results

solomon-negusse committed:
write result to s3

solomon-negusse committed:
minor fix

solomon-negusse committed:
return jsend format response;
return list of failed geometries;
clean up response not including jsend from individual lambda runs

solomon-negusse committed:
add failure message

solomon-negusse committed:
debug failed geoms

solomon-negusse committed:
remove debugging stmt

solomon-negusse committed:
change s3 -> https for results url

solomon-negusse committed:
use presigned urls for results

solomon-negusse committed:
fix linter mishap

solomon-negusse committed:
better name for response props

solomon-negusse committed:
simulate failure - first attempt

solomon-negusse committed:
simulate failure - 2nd attempt

solomon-negusse committed:
add uncaught errors in failed list

solomon-negusse committed:
revert failure tests

solomon-negusse committed:
Merge pull request #85 from wri/feature/postprocess-otf-lists

Feature/postprocess otf lists

dmannarino committed:
Implement pre-processing lambda

solomon-negusse committed:
generate presigned url in data api

solomon-negusse committed:
Merge pull request #87 from wri/feature/postprocess-otf-lists

generate presigned url in data api

dmannarino committed:
Remove unnecessary rasterio lambda layer

dmannarino committed:
Bump shapely version

dmannarino committed:
Downgrade pandas, shapely in an effort to save space

dmannarino committed:
Remove possibly superfluous layer

dmannarino committed:
Try using a megalayer to save space

dmannarino committed:
In desperation, modify the Dockerfile

dmannarino committed:
Add broken-out geopandas, numpy layers

dmannarino committed:
Cleanup pre-processing handler

dmannarino committed:
Merge branch 'feature/otf_lists' into feature/pre-process_otf_lists

dmannarino committed:
Use provided s3client instead of boto3 directly

dmannarino committed:
Reverse some gratuitous churn

dmannarino committed:
Various fixes

dmannarino committed:
Put WKB geometry in hex form

dmannarino committed:
Check for multiple geojson sources being specified, add default for id_field

dmannarino committed:
Adjust S3 prefixes

dmannarino committed:
Merge pull request #86 from wri/feature/pre-process_otf_lists

Fill in pre-process lambda for bulk OTF feature

danscales committed:
Set MaxConcurrency to 100, overall timeout to 5 mins

We can adjust these limits later as needed.

dmannarino committed:
Use temporary lambda layers

dmannarino committed:
Add missing numpy layer to raster_analysis_fanout

dmannarino committed:
Fix typo

dmannarino committed:
Merge pull request #88 from wri/point_to_temp_layers

Use temporary lambda layers

danscales committed:
Small change to trigger redeploy.

jterry64 committed:
Add state machine ARN to output

jterry64 committed:
Fix minor issues

dmannarino committed:
Fix FIXMEs in list pre-processing lambda

dmannarino committed:
Change docker-compose to docker compose to ffix GHA

dmannarino committed:
Merge pull request #89 from wri/address_preproc_fixmes

Address pre-processing FIXMEs (mosty propagate pipeline bucket name)

jterry64 committed:
Return appropriate errors based on results

jterry64 committed:
Merge

jterry64 committed:
Merge pull request #90 from wri/feature/otf_list_errors

Return appropriate errors based on results
